bebraven/platform
# Allow hard-coding <select> and <option> in Content Editor HTML

### Description

**Task**: https://app.asana.com/0/1174274412967132/1198222893411568

This implements hard-coding `<select>` and its `<option>`s for the Impact Survey's 1-10 scale in the Content Editor HTML view.

### Motivation and Context

This is what the Impact Surveys currently look like:
<img width="517" alt="Screen Shot 2020-10-26 at 12 30 22 PM" src="https://user-images.githubusercontent.com/62911934/98882235-ad5d1e80-2440-11eb-8440-0e42cd73ada4.png">

The slider components we were using aren't accessible, so we're switching to another component, while trying to balance the amount of effort spent building the authoring experience (e.g., making it configurable vs. hard-coding it) and making the surveys easy to fill out for the fellows.

This PR is a first step in that it lets us create Impact Surveys even if we don't have an authoring experience in the Content Editor's **Design** tab built for launch. 

Note that in this PR, we're doing a lot of manual HTML manipulation. In addition to adding `<select>` and `<option>` manually, we need to set the `name` attribute on `<select>`, so we have a key for our `SurveySubmissionAnswers` key-value store.

@rshipp is adding ID generation in #446 that we can probably piggy-back off of in a future PR to generate the `name`s.

### Test

- Everywhere we want a drop-down, paste the following HTML in the **Code** tab:
```
<select>
<option value="">&nbsp;</option>
<option value="1">1</option>
<option value="2">2</option>
<option value="3">3</option>
<option value="4">4</option>
<option value="5">5</option>
<option value="6">6</option>
<option value="7">7</option>
<option value="8">8</option>
<option value="9">9</option>
<option value="10">10</option>
</select>
```
<img width="1792" alt="Screen Shot 2020-11-11 at 5 00 54 PM" src="https://user-images.githubusercontent.com/62911934/98882002-2ad45f00-2440-11eb-8b66-cfc8e2b7db50.png">

- Toggle to **Code** tab and see that the selector appears:
<img width="1792" alt="Screen Shot 2020-11-11 at 5 00 58 PM" src="https://user-images.githubusercontent.com/62911934/98882019-3758b780-2440-11eb-98e3-bce5cb46c398.png">

- **Save** as a _Survey_ and preview it, checking that the values are set in the HTML:
<img width="1792" alt="Screen Shot 2020-11-11 at 5 01 38 PM" src="https://user-images.githubusercontent.com/62911934/98882047-46d80080-2440-11eb-9829-809b377c5630.png">

- I made some content edits so it looked more like our Impact Survey and published using our Course Management page:
<img width="1792" alt="Screen Shot 2020-11-11 at 5 00 37 PM" src="https://user-images.githubusercontent.com/62911934/98882103-6707bf80-2440-11eb-8096-1ebd2794130e.png">

- Manually change the `name` attribute for each `<select>` in the HTML tab:
<img width="1792" alt="Screen Shot 2020-11-11 at 5 05 32 PM" src="https://user-images.githubusercontent.com/62911934/98882150-843c8e00-2440-11eb-9cf8-fd20049b3d7a.png">

- Save and Publish to Canvas through the Course Management page:
<img width="1792" alt="Screen Shot 2020-11-11 at 5 02 23 PM" src="https://user-images.githubusercontent.com/62911934/98882177-90285000-2440-11eb-872b-3b3f25e1a17b.png">

- **Submit** as a Fellow after selecting some values:
<img width="1792" alt="Screen Shot 2020-11-11 at 4 24 43 PM" src="https://user-images.githubusercontent.com/62911934/98882370-f0b78d00-2440-11eb-83ab-079fec4459c5.png">

- The only way to see the answers is in our DB right now, so I used the console to check the `SurveySubmission.last.answers`, note that the names and values for the inputs are set:
<img width="1788" alt="Screen Shot 2020-11-11 at 4 38 07 PM" src="https://user-images.githubusercontent.com/62911934/98880976-d334f400-243d-11eb-9181-df6ca339f3a5.png">

### Types of changes




### Checklist:
